code/modules/clothing/spacesuits/rig/rig_pieces.dm

```diff
@@ -50,6 +50,25 @@
 	retract_while_active = FALSE
 	spawn_tags = null
 
+/obj/item/clothing/suit/space/rig/handle_shield(mob/user, damage, atom/damage_source = null, mob/attacker = null, def_zone = null, attack_text = "the attack")
+	if(istype(damage_source, /obj/item/projectile/bullet))
+		var/obj/item/projectile/bullet/B = damage_source
+
+		var/chance = max(round(armor.getRating(ARMOR_BULLET) - B.armor_penetration),0)
+		if(!(def_zone in list(BP_CHEST, BP_GROIN)))
+			chance *= 1.5
+		if(B.starting && prob(chance))
+			visible_message(SPAN_DANGER("\The [attack_text] ricochets off [user]'s [src.name]!"))
+
+			var/new_x = 3*(user.x + pick(0, 0, 1, -1, 1, -1, 2, -2)) - 2*B.starting.x // user - (bullet-user)*2
+			var/new_y = 3*(user.y + pick(0, 0, 1, -1, 1, -1, 2, -2)) - 2*B.starting.y
+
```

Comment:
can this for example , come back at the  firer ? or starting position ? what about picking an adjaecent turf from starting and then doing get_turf_away_from_target_complex(src , starting_turf, some_distance) , this way it would angle with how close they are and always go behind.